app/src/components/integrations/modals/addLdapIntegrationModal/twoSteps/twoStepsFooter/twoStepsFooter.jsx

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2024 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect } from 'react';
+import { Button } from '@reportportal/ui-kit';
+import classNames from 'classnames/bind';
+import PropTypes from 'prop-types';
+import { defineMessages, useIntl } from 'react-intl';
+import Parser from 'html-react-parser';
+import RightArrowIcon from 'common/img/arrow-right-inline.svg';
+import styles from './twoStepsFooter.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  nextStep: {
+    id: 'TwoStepsFooter.nextStep',
+    defaultMessage: 'Next step',
+  },
+  saveAndClose: {
+    id: 'TwoStepsFooter.saveAndClose',
+    defaultMessage: 'Save & close',
+  },
+  create: {
+    id: 'TwoStepsFooter.create',
+    defaultMessage: 'Create',
+  },
+  discardChanges: {
+    id: 'TwoStepsFooter.discardChanges',
+    defaultMessage: 'Discard changes',
+  },
+  cancel: {
+    id: 'TwoStepsFooter.cancel',
+    defaultMessage: 'Cancel',
+  },
+  previousStep: {
+    id: 'TwoStepsFooter.previousStep',
+    defaultMessage: 'Previous step',
+  },
+});
+
+export const createTwoStepsFooter = (
+  stepNumber,
+  onStepNumberChange,
+  onStepChange,
+  onDiscard,
+  isEdit,
+  onForceConfirmChange,
+  handleSubmit,
+  forceConfirm,
+  closeModal,
+) => {
+  return (closeHandler) => (
+    <TwoStepsFooter
+      stepNumber={stepNumber}
+      onStepNumberChange={onStepNumberChange}
+      onStepChange={onStepChange}
+      onDiscard={onDiscard}
+      isEdit={isEdit}
+      onForceConfirmChange={onForceConfirmChange}
+      handleSubmit={handleSubmit}
+      forceConfirm={forceConfirm}
+      closeModal={closeModal}
+      closeHandler={closeHandler.closeHandler}
+    />
+  );
+};
+
+export const TwoStepsFooter = ({
+  stepNumber,
+  onStepNumberChange,
+  onStepChange,
+  onDiscard,
+  isEdit,
+  onForceConfirmChange,
+  handleSubmit,
+  forceConfirm,
+  closeModal,
+  closeHandler,
+}) => {
+  const { formatMessage } = useIntl();
+
+  useEffect(() => {
+    closeModal && closeHandler();
+  }, [closeModal]);
```

Comment:
Why do we need to automatically close modal once we receive the `closeModal` prop?

Reply:
To close the modal, after we click "save and close" on the first step